src/waveform/waveformwidgetfactory.cpp

```diff
@@ -432,6 +441,22 @@ void WaveformWidgetFactory::setZoomSync(bool sync) {
     }
 }
 
+void WaveformWidgetFactory::setDisplayBeatGrid(bool sync) {
+    m_beatGridEnabled = sync;
+    if (m_config) {
+        m_config->set(ConfigKey("[Waveform]", "beatGridLinesCheckBox"), ConfigValue(m_beatGridEnabled));
```

Comment:
No, keep it consistent. Either leave it how it is or refactor how all the settings in this function are set.